hazelcast/src/main/java/com/hazelcast/sql/impl/partitioner/SingleValuePartitioner.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.partitioner;
+
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.sql.impl.row.Row;
+
+import java.io.IOException;
+
+/**
+ * Partitioner that is not making actual partitioning mapping all rows to the same partition.
+ */
+public final class SingleValuePartitioner implements RowPartitioner {
```

Comment:
`SingleValuePartitioner` is not a very correct name for this class, because there is no `value`. Maybe it should be something like `ZeroPartitioner`, `SinglePartitionPartitioner` or so, to better reflect the purpose.

Reply:
It partitions everything into the same partition, and that is where the single value comes from. But I'll rename.

Reply:
Renamed in https://github.com/hazelcast/hazelcast/pull/18013/commits/df15a9460ab87640faf70aaec530a8115cba9176